src/ai/backend/manager/api/vfolder.py

```diff
@@ -1819,23 +1843,55 @@ async def leave(request: web.Request, row: VFolderRow) -> web.Response:
     if row["is_owner"]:
         raise InvalidAPIParameters("Cannot leave a vfolder owned by the requesting user.")
 
-    root_ctx: RootContext = request.app["_root.context"]
     access_key = request["keypair"]["access_key"]
-    user_uuid = request["user"]["uuid"]
-    vfolder_id = row["id"]
-    perm = None
+    perm = row["permission"]
+    params = {
+        "vfolder_id": row["id"],
+        "user_uuid": request["user"]["uuid"],
+    }
     log.info(
-        "VFOLDER.LEAVE(ak:{}, vfid:{}, uid:{}, perm:{})", access_key, vfolder_id, user_uuid, perm
+        "VFOLDER.LEAVE(ak:{}, vfid:{}, uid:{}, perm:{})",
+        access_key,
+        params["vfolder_id"],
+        params["user_uuid"],
+        perm,
     )
-    async with root_ctx.db.begin() as conn:
-        query = (
-            sa.delete(vfolder_permissions)
-            .where(vfolder_permissions.c.vfolder == vfolder_id)
-            .where(vfolder_permissions.c.user == user_uuid)
-        )
-        await conn.execute(query)
-    resp = {"msg": "left the shared vfolder"}
-    return web.json_response(resp, status=200)
+    return await leave_request(request, params)
+
+
+@superadmin_required
+@server_status_required(ALL_ALLOWED)
+@vfolder_permission_required(VFolderPermission.READ_ONLY)
+@check_api_params(
+    t.Dict(
+        {
+            tx.AliasedKey(["shared_user_id", "sharedUserId"]): t.String,
+        }
+    ),
+)
+async def force_leave(request: web.Request, params: Any, row: VFolderRow) -> web.Response:
```

Comment:
Could we make "force" as an option to the existing API instead of adding a new API?